src/irmin-pack/unix/traverse_pack_file.ml

```diff
@@ -298,34 +351,46 @@ end = struct
     loop_entries ~buffer_off:0 Int63.zero None
 
   let run_or_test ~initial_buffer_size mode config =
+    let always =
+      let indexing_strategy = Conf.indexing_strategy config in
+      (* Using physical equality to test which indexing_strategy
+         we are using. Might not be great in the long term. *)
+      indexing_strategy == Irmin_pack.Indexing_strategy.always
```

Comment:
Could also use `Indexing_strategy.is_minimal`. It also uses physical equality but at least uses the modules declared equality check function.

Reply:
yes, thanks, didn't see that function